cleanup/src/main/scala/com/broadinstitute/dsp/cleanup/PubsubTopicAndSubscriptionCleaner.scala

```diff
@@ -0,0 +1,73 @@
+package com.broadinstitute.dsp
+package cleanup
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect.{Concurrent, Timer}
+import cats.mtl.Ask
+import com.google.pubsub.v1.{ProjectSubscriptionName, TopicName}
+import fs2.Stream
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.google2.{GoogleSubscriptionAdmin, GoogleTopicAdmin}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+class PubsubTopicAndSubscriptionCleaner[F[_]: Timer](config: PubsubTopicCleanerConfig,
+                                                     topicAdmin: GoogleTopicAdmin[F],
+                                                     subcriptionAdmin: GoogleSubscriptionAdmin[F],
+                                                     metrics: OpenTelemetryMetrics[F])(
+  implicit F: Concurrent[F],
+  logger: Logger[F]
+) {
+  def run(isDryRun: Boolean): F[Unit] = {
+    val res = for {
+      now <- Stream.eval(Timer[F].clock.realTime(TimeUnit.MILLISECONDS))
+      traceId = TraceId(s"pubsubTopicCleaner-${now}")
+      implicit0(ev: Ask[F, TraceId]) = Ask.const[F, TraceId](traceId)
+      deleteTopicStream = for {
+        topic <- topicAdmin.list(config.googleProject)
+        topicName = TopicName.parse(topic.getName)
+        _ <- if (topicName.getTopic.startsWith("leonardo-pubsub-") || topicName.getTopic.startsWith(
+                   "hamm-metadata-topic-fiab"
+                 )) {
+          if (isDryRun)
+            Stream.eval(logger.info(s"${topicName} will be deleted if run in nonDryRun mode"))
+          else
+            Stream.eval(metrics.incrementCounter("pubsubTopicAndSubscriptionCleaner")) >> Stream.eval(
+              topicAdmin.delete(topicName, Some(traceId))
+            )
+        } else Stream.eval(F.unit)
+      } yield ()
+      deleteSubscriptionStream = for {
+        subscription <- subcriptionAdmin.list(config.googleProject)
+        _ <- if (subscription.getTopic == "_deleted-topic_") { //will be `_deleted-topic_` if the topic has been deleted
```

Comment:
It took me some time to figure out that the `_deleted-topic_` guarantee comes from Google's [Subscription](https://googleapis.dev/java/google-cloud-pubsub/latest/com/google/pubsub/v1/Subscription.html) so it may be worth explicitly mentioning it.